File: app/src/main/java/jp/gr/java_conf/foobar/testmaker/service/views/PlayReviewView.java
package jp.gr.java_conf.foobar.testmaker.service.views;

import android.content.Context;
import android.support.annotation.Nullable;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.LinearLayout;
import android.widget.TextView;

import jp.gr.java_conf.foobar.testmaker.service.R;

public class PlayReviewView extends LinearLayout{

    TextView textAnswer;
    TextView textExplanation;

    Context context;

    public PlayReviewView(Context context) {
        super(context);
    }

    public PlayReviewView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);

        this.context = context;

        View layout = LayoutInflater.from(context).inflate(R.layout.layout_play_review,this);

        textAnswer = layout.findViewById(R.id.text_answer);

        textExplanation = layout.findViewById(R.id.text_explanation);
    }

    public PlayReviewView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public void setTextAnswer(String answer){

        textAnswer.setText(context.getString(R.string.message_answer,answer));

    }

    public void setTextExplanation(String explanation){

        textExplanation.setVisibility(VISIBLE);

        textExplanation.setText(context.getString(R.string.explanation, explanation));

        if(explanation.equals("")) textExplanation.setVisibility(GONE);

    }

}

File: app/src/main/java/jp/gr/java_conf/foobar/testmaker/service/view/play/PlayReviewView.kt
package jp.gr.java_conf.foobar.testmaker.service.view.play

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.view.View
import android.widget.LinearLayout
import jp.gr.java_conf.foobar.testmaker.service.R
import kotlinx.android.synthetic.main.layout_play_review.view.*

class PlayReviewView : LinearLayout {

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    init { LayoutInflater.from(context).inflate(R.layout.layout_play_review, this) }

    fun setTextAnswer(answer: String) { text_answer.text = context.getString(R.string.message_answer, answer) }

    fun setTextExplanation(explanation: String) {

        text_explanation.visibility = View.VISIBLE

        text_explanation.text = context.getString(R.string.explanation, explanation)

        if (explanation == "") text_explanation.visibility = View.GONE

    }

}
